product_pricelist_supplierinfo/models/product_pricelist.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2018 Tecnativa - Vicent Cubells
+# Copyright 2018 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class ProductPricelist(models.Model):
+    _inherit = "product.pricelist"
+
+    def _compute_price_rule(self, products, qty, uom=None, date=False, **kwargs):
+        """Recompute price after calling the atomic super method for
+        getting proper prices when based on supplier info.
+        """
+        rule_obj = self.env["product.pricelist.item"]
+        result = super()._compute_price_rule(products, qty, uom, date)
```

Comment:
You need to pass all params

```suggestion
        result = super()._compute_price_rule(products, qty, uom, date, **kwargs)
```